amplify/backend/function/fetchMotionState/src/utils.js

```diff
@@ -285,6 +285,7 @@ const updateMotionMessagesInDB = async (motionData, motionMessages, flag) => {
   const updatedStateHistory = {
     ...motionStateHistory,
     [flag]: true,
+    endedAt: flag === 'hasFailed' || flag === 'hasPassed' || flag === 'hasFailedNotFinalizable' ? new Date().toISOString() : null,
```

Comment:
Really wish we had typescript and `enum`s in here

Reply:
Same here 🙏